x-pack/filebeat/input/gcs/input_stateless.go

```diff
@@ -77,8 +78,9 @@ func (in *statelessInput) Run(inputCtx v2.Context, publisher stateless.Publisher
 		st := newState()
 		currentSource := source.(*Source)
 		log := inputCtx.Logger.With("project_id", currentSource.ProjectId).With("bucket", currentSource.BucketName)
-		metrics := newInputMetrics(inputCtx.ID+":"+currentSource.BucketName, nil)
-		defer metrics.Close()
+		// use a new metrics registry associated to no parent. No metrics will
+		// be published.
+		metrics := newInputMetrics(monitoring.NewRegistry())
```

Comment:
Another one replacing a nonempty id with an anonymous registry

Reply:
same as for the azureblobstorage, it's for test and the context has no metrics registry